lib/evmone/instructions_calls.cpp

```diff
@@ -226,9 +226,33 @@ Result create_eof_impl(
         if (initcode.data == nullptr)
         {
             stack.push(0);
-            return {EVMC_SUCCESS, gas_left};
+            return {EVMC_SUCCESS, gas_left};  // "Light" failure
         }
         initcontainer = {initcode.data, initcode.size};
+
+        // Charge for initcode validation.
+        const auto initcode_word_cost_validation = 2;
+        const auto initcode_cost_validation =
+            num_words(initcontainer.size()) * initcode_word_cost_validation;
+        if ((gas_left -= initcode_cost_validation) < 0)
+            return {EVMC_OUT_OF_GAS, gas_left};
+
+        const auto error_subcont = validate_eof(state.rev, initcontainer);
+
+        if (error_subcont != EOFValidationError::success)
+        {
+            stack.push(0);
+            return {EVMC_SUCCESS, gas_left};  // "Light" failure.
+        }
+
+        const auto initcontainer_header = read_valid_eof1_header(initcontainer);
+        const auto size_from_header =
+            static_cast<size_t>(initcontainer_header.data_offset + initcontainer_header.data_size);
```

Comment:
I would probably make this calculation a helper member of `EOF1Header`

Reply:
ah I see it was like this in the old code, fine to leave it like this then, but you could also move it

Reply:
I noticed that the reason and logic of this condition is a bit obscure, so tried moving it together to a helper `EOF1Header::can_init`. Looks much more readable this way